yarn-project/simulator/src/public/public_execution_context.ts

```diff
@@ -98,7 +98,23 @@ export class PublicExecutionContext extends TypedOracle {
    * @param args - Arguments to pack
    */
   public packArguments(args: Fr[]): Promise<Fr> {
-    return Promise.resolve(this.packedArgsCache.pack(args));
+    return Promise.resolve(this.packedValuesCache.pack(args));
+  }
+
+  /**
+   * Pack the given returns.
+   * @param returns - Returns to pack
+   */
+  public packReturns(returns: Fr[]): Promise<Fr> {
+    return Promise.resolve(this.packedValuesCache.pack(returns));
+  }
+
+  /**
+   * Unpack the given returns.
+   * @param returnsHash - Returns hash to unpack
+   */
+  public unpackReturns(returnsHash: Fr): Promise<Fr[]> {
+    return Promise.resolve(this.packedValuesCache.unpack(returnsHash));
```

Comment:
These two oracles will in principle not be supported by the AVM, so they should not be used in public/AVM. Please confirm with @dbanks12 or @Maddiaa0 how to proceed.

Reply:
They are not used in the AVM, only in functions marked with aztec(public)

Reply:
packReturns is only necessary in functions that get their return values swapped with some PublicInputs struct, so aztec(private) and aztec(public). Same as `packArguments` is present for public execution context

Reply:
> They are not used in AVM functions, only in functions marked with aztec(public)

Ah, you are right, I got confused because Avmis using things from public_context (FunctionReturns).

As long as there are no new oracles that are expected to work in AVM flows, I'm good with the PR.

Reply:
Yep no oracle has been added that needs to run in the AVM. The only thing is that the AVM context (as well as the public context) wraps the returned field array into a FunctionReturns for easy conversion into other types (`deserialize_into`)